modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/subscription/StorageQueue.java

```diff
@@ -414,17 +414,28 @@ public int clearMessagesReadToBufferForDelivery() {
     }
 
     /**
-     * Purge the messages in the queue. This will remove all the messages from read buffer.
+     * Purge all messages in the queue including messages in the store and slots for the queue.
      *
      * @return the number if messages removed from the read buffer
      */
-    public int purgeMessages() {
+    public int purgeMessages() throws AndesException {
         lastPurgedTimestamp = System.currentTimeMillis();
         log.info("Purging messages of queue " + name);
         return messageHandler.purgeMessagesOfQueue();
     }
 
     /**
+     * Purge the messages in the queue. This will remove all the messages from read buffer.
+     *
```

Comment:
What about comment "Purge only in-memory messages of the queue. This will remove all the messages from read buffer" ?